datafusion/expr/src/udf.rs

```diff
@@ -300,21 +308,32 @@ where
 /// # use std::any::Any;
 /// # use arrow::datatypes::DataType;
 /// # use datafusion_common::{DataFusionError, plan_err, Result};
-/// # use datafusion_expr::{col, ColumnarValue, Signature, Volatility};
+/// # use datafusion_expr::{col, ColumnarValue, Documentation, Signature, Volatility};
 /// # use datafusion_expr::{ScalarUDFImpl, ScalarUDF};
+/// # use datafusion_expr::scalar_doc_sections::DOC_SECTION_MATH;
+///
 /// #[derive(Debug)]
 /// struct AddOne {
-///   signature: Signature
+///   signature: Signature,
 /// }
 ///
 /// impl AddOne {
 ///   fn new() -> Self {
 ///     Self {
-///       signature: Signature::uniform(1, vec![DataType::Int32], Volatility::Immutable)
+///       signature: Signature::uniform(1, vec![DataType::Int32], Volatility::Immutable),
 ///      }
 ///   }
 /// }
-///
+///  
+/// const DOCUMENTATION: Documentation = Documentation {
+///     doc_section: DOC_SECTION_MATH,
+///     description: "Add one to an int32",
+///     syntax_example: "add_one(2)",
+///     sql_example: None,    
+///     arguments: Some(&[("arg_1", "The int32 number to add one to")]),
+///     related_udfs: None,
+/// };
```

Comment:
Perhaps we can add a Builder style API (in a follow on PR) to make these examples look more like

```rust
Documentation::builder()
    .with_doc_section(DOC_SECTION_MATH)
    .with_description("Add one to an int32")
    .with_syntax_example("add_one(2)")
    .with_argument("arg_1", "The int32 number to add one to")
    .build()
```

Reply:
Done, and used for all of the implemented udf's. Much nicer!